cmd/cosign/cli/sign/sign_blob.go

```diff
@@ -33,9 +36,28 @@ import (
 	"github.com/sigstore/cosign/v2/pkg/cosign"
 	cbundle "github.com/sigstore/cosign/v2/pkg/cosign/bundle"
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
+	"github.com/sigstore/sigstore/pkg/signature"
 	signatureoptions "github.com/sigstore/sigstore/pkg/signature/options"
 )
 
+func getHashAlgorithmFromSignerVerifier(sv *SignerVerifier) (crypto.Hash, error) {
+	publicKey, err := sv.SignerVerifier.PublicKey()
+	if err != nil {
+		return crypto.Hash(0), err
+	}
+
+	switch publicKey.(type) {
+	case *ecdsa.PublicKey:
+		return crypto.SHA256, nil
+	case *rsa.PublicKey:
+		return crypto.SHA256, nil
+	case ed25519.PublicKey:
+		return crypto.SHA512, nil
+	default:
+		return crypto.Hash(0), fmt.Errorf("unsupported public key type")
+	}
+}
```

Comment:
I believe this would be solved with https://github.com/sigstore/sigstore/issues/860 . Shall I add a reference to this issue in the code (and back-ref)?

Reply:
Would you be interested in taking a look at the failing test in https://github.com/sigstore/sigstore/pull/1426? I think it's just a failure due to a mock not having an expectation met. We could get that merged in then to use here.

Reply:
Isn't https://github.com/sigstore/sigstore/pull/1426 just about the KMS SignerVerifier? I think we'd need that `HashFunc` feature at the `sigstore.SignerVerifier` level.